sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -377,8 +382,13 @@ def process(self, element, *args, **kwargs):
     if not source_list:
       return
     source = source_list[0].source
-    for record in source.read(range.new_tracker()):
-      yield record
+
+    if self._with_context:
```

Comment:
Probably cleaner to write as:

for record in source.read(range.new_tracker()):
  if self._with_context:
    ...
  else:
    ...

Reply:
The reason I didn't do it like that was to avoid checking for "self._with_context" for all records. I refactored this, please let me know if that would work. Thanks